### PR TITLE
User authority functional #4503

### PR DESCRIPTION
**Pull request by innayashna**
User authority functional added

### CODE REVIEW

**Review by danylohlynsky — APPROVED** [review]

**Review by NataliaHlazova — APPROVED** [review]

File: core/src/test/java/greencity/ModelUtils.java [hunk 1]
```diff
@@ -652,4 +653,13 @@ public static CreateAddressRequestDto getAddressRequestDto() {
             .entranceNumber("3")
             .build();
     }
+
+    public static UserEmployeeAuthorityDto getUserEmployeeAuthorityDto() {
+        List<String> authorities = new ArrayList<>();
+        authorities.add("SEE_CLIENTS_PAGE");
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/ModelUtils.java]: Please move `SEE_CLIENTS_PAGE` to constant

File: core/src/test/java/greencity/ModelUtils.java [hunk 2]
```diff
@@ -652,4 +653,13 @@ public static CreateAddressRequestDto getAddressRequestDto() {
             .entranceNumber("3")
             .build();
     }
+
+    public static UserEmployeeAuthorityDto getUserEmployeeAuthorityDto() {
+        List<String> authorities = new ArrayList<>();
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/ModelUtils.java]: Use `Collections.singletonList("SEE_CLIENTS_PAGE")`

File: core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java [hunk 3]
```diff
@@ -198,4 +204,30 @@ void deletePosition() throws Exception {
         mockMvc.perform(delete(UBS_LINK + DELETE_POSITION_LINK + "/1").principal(principal)).andExpect(status().isOk());
         verify(service, times(1)).deletePosition(1L);
     }
+
+    @Test
+    void getAllAuthorities() throws Exception {
+        Set<String> authorities = new HashSet<>();
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java:210]: Use `Collections.singleton("ADMIN")`

File: core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java [hunk 4]
```diff
@@ -198,4 +204,30 @@ void deletePosition() throws Exception {
...
+        authorities.add("ADMIN");
+        when(ubsClientService.getAllAuthorities(anyString())).thenReturn(authorities);
+        mockMvc.perform(get(UBS_LINK + "/get-all-authorities" + "?email=test@mail.com"))
+            .andExpect(status().isOk());
+        verify(ubsClientService, times(1)).getAllAuthorities("test@mail.com");
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java]: `verify(ubsClientService) `is the same as `verify(ubsClientService, times(1))`

File: core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java [hunk 5]
```diff
@@ -198,4 +204,30 @@ void deletePosition() throws Exception {
...
+        String json = objectMapper.writeValueAsString(dto);
+
+        mockMvc.perform(put(UBS_LINK + "/edit-authorities")
+            .principal(principal)
+            .content(json)
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk());
+
+        String email = principal.getName();
+        verify(ubsClientService, times(1)).updateEmployeesAuthorities(dto, email);
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/controller/ManagementEmployeeControllerTest.java]: same here

File: service-api/src/main/java/greencity/dto/employee/UserEmployeeAuthorityDto.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+import java.util.List;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class UserEmployeeAuthorityDto {
+    private Long employeeId;
+    private List<String> authorities;
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/dto/employee/UserEmployeeAuthorityDto.java:16]: Could authorities be an `enum`?

File: service-api/src/main/java/greencity/service/ubs/UBSClientService.java [hunk 7]
```diff
@@ -21,10 +22,7 @@
 import greencity.exceptions.payment.PaymentLinkException;
 import org.springframework.data.domain.Pageable;
 
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/service/ubs/UBSClientService.java]: please do not use `*`

File: service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java [hunk 8]
```diff
@@ -3,20 +3,12 @@
...
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java]: same here

**Review by OlegVat — APPROVED** [review]

**Pull request merged**